cpp/src/parquet/statistics.cc

```diff
@@ -792,6 +805,8 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
     this->statistics_.null_count = 0;
     this->statistics_.distinct_count = 0;
     this->num_values_ = 0;
+    this->statistics_.is_min_value_exact = std::nullopt;
+    this->statistics_.is_max_value_exact = std::nullopt;
```

Comment:
I'm not sure why these are reset here, since this method is ostensibly resetting the various counts, not the extrema.